src/main/scala/io/citrine/lolo/trees/multitask/MultiTaskTrainingNode.scala

```diff
@@ -43,40 +76,36 @@ class MultiTaskTrainingNode(
     val (left, right) = reducedData.partition(r => split.turnLeft(r._1))
 
     // Construct an internal node if the children are defined and actually have valid data
-    val node = if (leftChild.isDefined && rightChild.isDefined && left.nonEmpty && right.nonEmpty) {
-      // Not a case because of erasure :-(
-      if (label.isInstanceOf[Double]) {
-        new InternalModelNode[PredictionResult[Double]](
-          split,
-          leftChild.get.getNode(index).asInstanceOf[ModelNode[PredictionResult[Double]]],
-          rightChild.get.getNode(index).asInstanceOf[ModelNode[PredictionResult[Double]]],
-          numFeatures=inputs.head._1.length,
-          outputDimension = 0,  // Don't support multitask SHAP at this time.
-          trainingWeight = reducedData.length.toDouble
-        )
-      } else {
-        if (!label.isInstanceOf[Char]) throw new IllegalArgumentException("Training data wasn't double or char")
-        new InternalModelNode[PredictionResult[Char]](
-          split,
-          leftChild.get.getNode(index).asInstanceOf[ModelNode[PredictionResult[Char]]],
-          rightChild.get.getNode(index).asInstanceOf[ModelNode[PredictionResult[Char]]],
-          numFeatures=inputs.head._1.length,
-          outputDimension = 0,  // Don't support multitask SHAP at this time.
-          trainingWeight = reducedData.length.toDouble
-        )
-      }
-    } else if (leftChild.isDefined && left.nonEmpty) {
-      leftChild.get.getNode(index)
-    } else if (rightChild.isDefined && right.nonEmpty) {
-      rightChild.get.getNode(index)
-    } else {
-      // If there are no children or the children don't have valid data for this label, emit a leaf (with GTM)
-      if (label.isInstanceOf[Double]) {
-        new TrainingLeaf[Double](reducedData.asInstanceOf[Seq[(Vector[AnyVal], Double, Double)]], new GuessTheMeanLearner(), 1).getNode()
-      } else {
-        new TrainingLeaf[Char](reducedData.asInstanceOf[Seq[(Vector[AnyVal], Char, Double)]], new GuessTheMeanLearner(), 1).getNode()
-      }
+    (leftChild, rightChild) match {
```

Comment:
this is just a refactoring from an if-else block to a match, which avoids the need for `.get` statements